src/components/CategoryModal.tsx

```diff
@@ -2,33 +2,46 @@ import { FC } from 'react'
 import { Form } from 'react-router-dom'
 
 interface ICategoryModal {
-  type: 'post' | 'patch',
-  id: number,
-  setVisibleModal: (visible: boolean) => void
+	type: 'post' | 'patch'
+	id?: number
+	setVisibleModal: (visible: boolean) => void
 }
 
-const CategoryModal: FC<ICategoryModal> = ({type, id, setVisibleModal}) => {
+const CategoryModal: FC<ICategoryModal> = ({ type, id, setVisibleModal }) => {
+	return (
+		<div className="fixed top-0 bottom-0 left-0 right-0 w-full h-full bg-black/50 flex justify-center items-center">
+			<Form
+				action="/categories"
+				method={type}
+				onSubmit={() => setVisibleModal(false)}
+				className="grid gap-2 w-[300px] p-5 rounded-md bg-slate-900"
+			>
+				<label htmlFor="title">
+					<small>Category Title</small>
+					<input
+						className="input w-full"
+						type="text"
+						name="title"
+						placeholder="Title..."
+					/>
+					<input type="hidden" name="id" value={id} />
+				</label>
 
-  return (
-    <div className='fixed top-0 bottom-0 left-0 right-0 w-full h-full bg-black/50 flex justify-center items-center'>
-      <Form className='grid gap-2 w-[300px] p-5 rounded-md bg-slate-900'>
-        <label htmlFor="title">
-          <small>Category Title</small>
-          <input className='input w-full' type="text" name='title' placeholder='Title...' />
-        </label>
-
-        <div className='flex items-center gap-2'>
-          <button className='btn btn-green ' type='submit'>
-            { type === 'patch' ? 'Save' : 'Create'}
-            Submit
-          </button>
-          <button className='btn btn-red '>
-            Close
-          </button>
-        </div>
-      </Form>
-    </div>
-  )
+				<div className="flex items-center gap-2">
+					<button className="btn btn-green " type="submit">
+						{type === 'patch' ? 'Save' : 'Create'}
+						Submit
```

Comment:
Not a big deal, but this creates a button with text "SaveSubmit" or "CreateSubmit". THe previous line takes care of the button text